main.go

```diff
@@ -142,12 +144,32 @@ func main() {
 	if err != nil {
 		os.Exit(1)
 	}
-
-	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+	checker := healthz.Ping
+	// Setup webhooks if requested
+	if strings.ToLower(os.Getenv("ENABLE_WEBHOOKS")) != "false" {
+
+		if err = (&watcherv1beta1.Watcher{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "Watcher")
+			os.Exit(1)
+		}
+		if err = (&watcherv1beta1.WatcherAPI{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "WatcherAPI")
+			os.Exit(1)
+		}
+		if err = (&watcherv1beta1.WatcherApplier{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "WatcherApplier")
+			os.Exit(1)
+		}
+		if err = (&watcherv1beta1.WatcherDecisionEngine{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "WatcherDecisionEngine")
+			os.Exit(1)
+		}
```

Comment:
Should add a line like this [1] after this line.

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/80d688a88c19a7cc99ffae4f227924a7810db107/main.go#L161

Reply:
Done!

Reply:
this is somethign i never got around to optimising in the nova operator but 
again i would like to clean this up into a for loop over like we do with 

NewReconcilers

https://github.com/openstack-k8s-operators/watcher-operator/blob/main/controllers/watcher_common.go#L56-L79

we have one function that map of reconsiler name to and instance of the object

and then a second function that struct and loops over each setting them up and checking for an error.

https://github.com/openstack-k8s-operators/watcher-operator/blob/main/controllers/watcher_common.go#L81-L91


we should try and avoid having long if chains like this were we can by transforming it into a data-driven flow.

that is escpially true when it all or nothing (if any fails we abort)
and when it needs to be done in two places, main.go and suite_tests.go

Reply:
I am still working on this piece.

Reply:
sure lets move forward with this one as is and follow up